File: src/main/java/com/koddy/server/member/domain/model/response/ScheduleResponse.java
package com.koddy.server.member.domain.model.response;

import com.koddy.server.member.domain.model.mentor.Timeline;

public record ScheduleResponse(
        String dayOfWeek,
        Start start,
        End end
) {
    public record Start(
            int hour,
            int minute
    ) {
    }

    public record End(
            int hour,
            int minute
    ) {
    }

    public static ScheduleResponse from(final Timeline timeline) {
        return new ScheduleResponse(
                timeline.getDayOfWeek().getKor(),
                new ScheduleResponse.Start(timeline.getStartTime().getHour(), timeline.getStartTime().getMinute()),
                new ScheduleResponse.End(timeline.getEndTime().getHour(), timeline.getEndTime().getMinute())
        );
    }
}

File: src/main/java/com/koddy/server/member/domain/model/response/ScheduleResponse.kt
package com.koddy.server.member.domain.model.response

import com.koddy.server.member.domain.model.mentor.Timeline

data class ScheduleResponse(
    val dayOfWeek: String,
    val start: Start,
    val end: End,
) {
    data class Start(
        val hour: Int,
        val minute: Int,
    )

    data class End(
        val hour: Int,
        val minute: Int,
    )

    companion object {
        fun from(timeline: Timeline): ScheduleResponse =
            ScheduleResponse(
                dayOfWeek = timeline.dayOfWeek.kor,
                start = Start(
                    hour = timeline.startTime.hour,
                    minute = timeline.startTime.minute,
                ),
                end = End(
                    hour = timeline.endTime.hour,
                    minute = timeline.endTime.minute,
                ),
            )
    }
}
